src/solana/SolanaEngine.ts

```diff
@@ -726,12 +726,12 @@ export class SolanaEngine extends CurrencyEngine<
         lamports: parseInt(nativeAmount)
       })
       instructions.push(transferInstruction)
-    }
 
-    if (tokenId != null && eq(totalTxAmount, balance)) {
-      // This is a max token send so we don't need to consider the minimumAddressBalance
-    } else if (gt(add(totalTxAmount, minimumAddressBalance), balance)) {
-      throw new InsufficientFundsError({ tokenId })
+      if (eq(totalTxAmount, balance)) {
```

Comment:
So this check is only relevant to SOL, not tokens? Because before, it only applied to tokens for some reason. I guess that was the bug.

Reply:
Yeah this specific minimumAddressBalance check is only relevant to SOL transfers and was incorrectly checking token amounts